### ISSUE TITLE
BUG Split with start & end keys

### ISSUE THREAD

**Issue by 07ke** [Psychopoulet/split-frames#32]
```ts
 const stream = new Readable({
      read() {}
    })

    stream
      .pipe(
        new Splitter({
          startWith: Buffer.from([0x28, 0x54, 0x59, 0x50, 0x45, 0x3a]),
          endWith: 0x29
        })
      )
      .on('data', (chunk: Buffer) => {
        Logger.log('ack received', chunk.toString()) // (only 1x) -> good, escaped, in data
      })

    // Yes
    stream.push(Buffer.from([0x28, 0x54, 0x59, 0x50, 0x45, 0x3a, 0x30, 0x31, 0x31, 0x30, 0x29]))

    // No
    stream.push(Buffer.from([0x28]))
    stream.push(Buffer.from([0x54, 0x59, 0x50, 0x45, 0x3a, 0x30, 0x31, 0x31, 0x30, 0x29]))

    // No
    stream.push(Buffer.from([0x28, 0x54]))
    stream.push(Buffer.from([0x59, 0x50, 0x45, 0x3a, 0x30, 0x31, 0x31, 0x30, 0x29]))

    // No
    stream.push(Buffer.from([0x28, 0x54, 0x59]))
    stream.push(Buffer.from([0x50, 0x45, 0x3a, 0x30, 0x31, 0x31, 0x30, 0x29]))

    // Yes
    stream.push(Buffer.from([0x28, 0x54, 0x59, 0x50, 0x45, 0x3a]))
    stream.push(Buffer.from([0x30, 0x31, 0x31, 0x30, 0x29]))
```

**Comment by Psychopoulet** [comment]
oh, you're right, I'll correct it in the next patch version & add your samples as unit tests.
thanks !

**Comment by Psychopoulet** [comment]
corrected in 2.5.1
